packages/zoe/src/contracts/autoswap.js

```diff
@@ -1,52 +1,44 @@
 /* eslint-disable no-use-before-define */
 import harden from '@agoric/harden';
-import Nat from '@agoric/nat';
 import produceIssuer from '@agoric/ertp';
 import { assert, details } from '@agoric/assert';
 
-import { natSafeMath } from './helpers/safeMath';
-import { makeHelpers } from './helpers/userFlow';
+import { makeZoeHelpers } from './helpers/zoeHelpers';
 import { makeConstProductBC } from './helpers/bondingCurves';
 
-export const makeContract = harden((zoe, terms) => {
-  // The user passes in an array of two issuers for the two kinds of
-  // assets to be swapped.
-  const startingIssuers = terms.issuers;
+// Autoswap is a rewrite of Uniswap. Please see the documentation for
+// more https://agoric.com/documentation/zoe/guide/contracts/autoswap.html
 
-  // There is also a third issuer, the issuer for the liquidity token,
-  // which is created in this contract. We will return all three as
-  // the canonical array of issuers for this contract
+export const makeContract = harden(zoe => {
+  // Create the liquidity mint and issuer.
   const { mint: liquidityMint, issuer: liquidityIssuer } = produceIssuer(
     'liquidity',
   );
-  const issuers = [...startingIssuers, liquidityIssuer];
 
-  const LIQ_INDEX = 2;
   let poolHandle;
   let liqTokenSupply = 0;
 
-  const { subtract } = natSafeMath;
-
-  return zoe.addNewIssuer(liquidityIssuer).then(() => {
-    const amountMathArray = zoe.getAmountMathForIssuers(issuers);
-    amountMathArray.forEach(amountMath =>
+  return zoe.addNewIssuer(liquidityIssuer, 'Liquidity').then(() => {
+    const { issuerKeywordRecord, keywords } = zoe.getInstanceRecord();
```

Comment:
`keywords` here will presumably be `TokenA`, `TokenB`, and `liquidity` in some order. On line 247...250 in `publicAPI` `getPrice` is
```js
            assert(
              keywords.includes(inKeyword),
              details`keyword ${inKeyword} was not valid`,
            );
```
That will pass if `inKeyword` is `liquidity`.